subtitles/vosk_transcriber.py

```diff
@@ -31,7 +36,12 @@ def generate(self, source: str, language: str) -> (str, str):
                                '-f', 'wav', "-"],
                               stdout=subprocess.PIPE).stdout as stream:
             if language in self.__recognizers:
-                return srt_to_vtt(self.__recognizers[language].SrtResult(stream)), language
+                mutex_vosk.acquire()
+                try:
+                    return self.__recognizers[language].SrtResult(stream), language
```

Comment:
You need to create recognizer here, not on init.

Reply:
Oh yeah that totally makes sense. I really was wondering why Vosk needed a mutex too and I couldn't figure out why. Thanks a lot!